docs/source/package-spec.rst

```diff
@@ -518,13 +505,69 @@ for the following common resources.
   - ``documentation``: Package Documentation
   - ``repository``: Location of the project source code.
 
-
   :Key: ``links``
 
   :Type: Object (String: String)
 
 ----
 
+.. _Sources Object:
+
+The *Sources* Object
+~~~~~~~~~~~~~~~~~~~~
+
+A *Sources* object is defined to have the following fields.
+
+  :Key: A global identifier for the source file. (string)
+  :Value: `SourceObject`_
+
+.. _SourceObject:
+
+Source Object
+^^^^^^^^^^^^^
+
+Checksum: ``checksum``
+^^^^^^^^^^^^^^^^^^^^^^
+
+Hash of the source file - prefixed by the hash algorithm and a ``/``.
+  - (eg. ``keccak256/abc123...``).
+
+  TODO: Can we enforce content-addressed urls and remove this field entirely?
+  :Required: If there are no urls that contain a hash of its contents.
+  :Key: ``checksum``
+  :Value: string
+
+URLs: ``urls``
+^^^^^^^^^^^^^^
+Array of urls that resolve to the same source file.
+  - Urls **should** be stored on a content-addressable filesystem.
+  - If the resulting document is a directory the key **should** be interpreted as a directory path.
+  - If the resulting document is a file the key **should** be interpreted as a file path.
```

Comment:
Do we want anything in here about the URLs requiring a proper `scheme`  i.e. we probably don't want: `QmPXME1oRtoT627YKaDPDQ3PwA8tdP9rWuAAweLzqSwAWT` but rather want `ipfs://QmPXME1oRtoT627YKaDPDQ3PwA8tdP9rWuAAweLzqSwAWT`